core/src/jsMain/kotlin/external/BluetoothAdvertisingEvent.kt

```diff
@@ -6,6 +6,11 @@ import org.w3c.dom.events.Event
  * https://webbluetoothcg.github.io/web-bluetooth/#bluetoothadvertisingevent
  */
 internal abstract external class BluetoothAdvertisingEvent : Event {
-    val rssi: Int
     val device: BluetoothDevice
+    val uuids: Array<String>
+    val name: String?
+    val rssi: Int?
+    val txPower: Int?
+    val manufacturerData: Any?
```

Comment:
The docs say this is map-like. Does that mean that we could just mark it as a `Map` in Kotlin? Since I assume the functions and properties would align with that of a typical map (assuming Kotlin and JavaScript have similar naming for their `Map` properties/functions)?

Wondering if we have access to `keys` and could ask for element at index `0` or use `first`, to be able to ultimately grab the manufacturer ID.

Reply:
The docs indicate they conform to `maplike` but I've found this to be only partially true for the current (experimental) implementation of both the service and manufacturer data. Typecasting to a `Map` in Kotlin fails as none of the expected methods are implemented.

The only methods that appear to be exposed are the `get` and a `forEach` iterator.

> having to provide a manufacturer ID is quite awkward for a library user

Perhaps, but the company identifiers are ratified in the BLE specification so are not difficult to find. Both the JS and Android framework designers also appear to agree this is their preferred approach. Apple disagrees but they often do lol. ¯\\_(ツ)_/¯ 

The JS code to obtain the `keys` would use the iterator to traverse the entries, very straightforward in native JS, somewhat convoluted from Kotlin directly.

Reply:
Bummer it's not a straightforward / easy cast to `Map`.
Rather than using `Any?` (and then later having to use `asDynamic`) could we make this `BluetoothManufacturerDataMap` type?

Then define `BluetoothManufacturerDataMap` as something similar to?:

```kotlin
internal interface BluetoothManufacturerDataMap {
    fun get(key: String): ByteArray
    fun forEach(action: (key: String, value: ByteArray) -> Unit): Unit
}
```

Reply:
Found a more generic mechanism by hooking into the javascript Object iterator interface and bridging it as a Kotlin `Iterable`. This lets us access any of the javascript methods that return `Iterator` such as `keys()` and `entries()` and typecast them as `Iterable`. This looks to be a less brittle approach and doesn't require any custom types.

See https://github.com/JuulLabs/kable/pull/41/commits/f6995fd5972c9931ba52079734ce8cad614e397e